crypto/s2n_fips.c

```diff
@@ -26,17 +26,24 @@
 
 static bool s2n_fips_mode_enabled = false;
 
-/* FIPS mode can be checked if OpenSSL was configured and built for FIPS which then defines OPENSSL_FIPS.
+/* FIPS mode can be checked if OpenSSL was configured and built for FIPS which
+ * then defines OPENSSL_FIPS.
  *
- * AWS-LC always defines FIPS_mode() that you can call and check what the library was built with. It does not define
- * a public OPENSSL_FIPS/AWSLC_FIPS macro that we can (or need to) check here
+ * AWS-LC always defines FIPS_mode() that you can call and check what the
+ * library was built with. It does not define a public OPENSSL_FIPS/AWSLC_FIPS
+ * macro that we can (or need to) check here
  *
- * Safeguard with macro's, for example because Libressl dosn't define
+ * Safeguard with macro's, for example because Libressl doesn't define
  * FIPS_mode() by default.
  *
- * Note: FIPS_mode() does not change the FIPS state of libcrypto. This only returns the current state. Applications
- * using s2n must call FIPS_mode_set(1) prior to s2n_init.
- * */
+ * Note: FIPS_mode() does not change the FIPS state of libcrypto. This only
+ * returns the current state. Applications using s2n must call FIPS_mode_set(1)
+ * prior to s2n_init.
+ *
+ * Note: Developers should use `s2n_is_in_fips_mode()` instead of calling this
+ * directly. `s2n_is_in_fips_mode()` returns libcrypto FIPS status at library
+ * initialization, ie. s2n_init().
+ */
```

Comment:
If we're going to rewrite this documentation, let's really clean it up instead of just tacking more "Note"s on :) How about:

> 
> This method should ONLY be called during library initialization (from s2n_init). This method indicates the state of the libcrypto, NOT the state of s2n-tls. For example, if s2n-tls is using Openssl and FIPS_mode_set(1) is called after s2n_init is called, then this method will return true while s2n_is_in_fips_mode() will return false and s2n-tls will not operate in FIPS mode.
> 
> For AWS-LC, the FIPS_mode() method is always defined. If AWS-LC was built to support FIPS, FIPS_mode() always returns 1.
> 
> For OpenSSL, OPENSSL_FIPS is defined if the libcrypto was built to support FIPS. The FIPS_mode() method is only present if OPENSSL_FIPS is defined, and only returns 1 if FIPS_mode_set(1) was used to enable FIPS mode before s2n_init was called.

Reply:
> and only returns 1 if FIPS_mode_set(1) was used to enable FIPS mode before s2n_init was called.

Is that true. I would expect `FIPS_mode()` to return 1 even if `FIPS_mode_set(1)` is called after s2n_init().

Reply:
Oh good point, probably drop the "before s2n_init was called" part. My first sentence literally warns about that case lol